manifests/scrape_job.pp

```diff
@@ -0,0 +1,42 @@
+# Define: prometheus::scrape_job
+#
+# This define is used to export prometheus scrape settings from nodes to be scraped to the node
+# running prometheus itself.
+# This can be used to make prometheus find instances of your running service or application.
+#
+#  [*job_name*]
+#  The name of the scrape job. This will be used when collecting resources on the prometheus node.
+#  Corresponds to the prometheus::collect_scrape_jobs parameter.
+#
+#  [*targets*]
+#  Array of hosts and ports in the form "host:port"
+#
+#  [*labels*]
+#  Labels added to the scraped metrics on the prometheus side, as label:values pairs
+#
+#  [*collect_dir*]
+#  Directory used for collecting scrape definitions.
+#  NOTE: this is a prometheus setting and will be overridden during collection.
+#
+define prometheus::scrape_job (
+  String $job_name,
+  Array[String] $targets,
+  Hash[String, String] $labels = {},
+  String $collect_dir          = undef,
+  Optional[String] $scheme     = 'http',
+) {
+  $config = to_yaml([
+    {
+      targets => $targets,
+      labels  => $labels,
+      scheme  => $scheme,
```

Comment:
Does this actually work? Can't find the `scheme` key in [the docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#file_sd_config).
The reason for my original - admittedly a bit unintuitive - list of hashes approach was precisely because the scheme seems to only exist in the  `scrape_config` context, not in the context of the individual `targets` (and therefore not in the context of the puppet `scrape_job` exported resource).
Did I miss something?

Reply:
ouch. you're probably right - i'll take a look.

Reply:
you're right, it does not work. i'll see if I can just reuse the original pattern... :/

Reply:
i've reverted to the origina data structure in #141 even though i don't quite like it. :)